File: JavaRepo/lab2/src/model/Statement/PrintStmt.java
package model.Statement;

import exceptions.DivideByZeroException;
import exceptions.InvalidOperandException;
import exceptions.NotExistingException;
import model.Expressions.Exp;
import utils.*;
import utils.adt.MyIDictionary;
import utils.adt.MyIHeap;
import utils.adt.MyIList;

public class PrintStmt implements IStmt {
    private Exp exp;

    public PrintStmt(Exp _exp) {
        exp = _exp;
    }

    @Override
    public String toString(){
        return "print(" + exp.toString() + ")";
    }

    @Override
    public PrgState execute(PrgState state) throws DivideByZeroException, InvalidOperandException, NotExistingException {
        MyIList<Integer> list = state.getOut();
        MyIDictionary<String, Integer> dict = state.getSymTable();
        MyIHeap<Integer, Integer> heap = state.getHeap();
        int res = exp.eval(dict, heap);
        list.add(res);
        return state;
    }
}

File: MAP/C# - Toy language/Lab10/Toy Language/Toy Language/model/statement/PrintStmt.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Toy_Language.model.expressions;
using Toy_Language.utils;

namespace Toy_Language.model.statement
{
    public class PrintStmt : IStmt
    {
        private Exp exp;

        public PrintStmt(
            Exp _exp
            )
        {
            exp = _exp;
        }

        public PrgState Execute(PrgState state)
        {
            IModList<int> list = state.OutList;
            IModDictionary<string, int> dict = state.SymbTable;

            int res = exp.Eval(dict);
            list.Add(res);

            return state;
        }

        public override string ToString()
        {
            return "print(" + exp.ToString() + ")";
        }

    }
}
